conjure-java-core/src/main/java/com/palantir/conjure/java/types/BeanBuilderGenerator.java

```diff
@@ -805,20 +829,13 @@ private boolean isByteBuffer(Type type) {
         return type.accept(TypeVisitor.IS_BINARY) && !options.useImmutableBytes();
     }
 
-    private List<MethodSpec> createAuxiliarySetters(
-            EnrichedField enriched,
-            Map<com.palantir.conjure.spec.TypeName, TypeDefinition> typesMap,
-            boolean override) {
+    private List<MethodSpec> createAuxiliarySetters(EnrichedField enriched, boolean override) {
         Type type = enriched.conjureDef().getType();
         Optional<LogSafety> safety = safetyEvaluator.getUsageTimeSafety(enriched.conjureDef());
 
         ImmutableList.Builder<MethodSpec> builder = ImmutableList.builder();
 
         if (type.accept(TypeVisitor.IS_LIST)) {
-            if (isPrimitiveOptimized(type)) {
```

Comment:
Hoisted this up as having it nested in the auxiliary setters was a hold over from a time before `isPrimitiveOptimized` properly guarded for collection types only. Hence it needing to be nested inside a `type.accept(TypeVisitor.IS_LIST)`